app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -227,9 +227,24 @@ def method_missing(method_name, *args, &block)
       end
     end
 
-    def find_mac_address_on_network(nics, network, log)
+    def get_mac_address_of_nic_on_requested_vlan(args)
+      network = find_network_in_cluster(args[:value_of_vlan_option], args[:dest_cluster], args[:source_template])
+      return nil if network.nil?
+
+      find_mac_address_on_network(network, args[:destination])
+    end
+
+    def find_network_in_cluster(network_name, dest_cluster, source)
+      network = source.ext_management_system.ovirt_services.cluster_find_network_by_name(dest_cluster.ems_ref, network_name)
```

Comment:
Because thia instance is initialized with the ems from the destination vm and that one is run on the ema from the source template. When running the code i saw the ems on the vm is not set yet (probably since the vm is only beimg provisioned) and it is required for this method to work so this actually initialzes an instance with ems of the source template and calls the method on it.